src/edu/macalester/conceptual/cli/CommandLine.java

```diff
@@ -226,6 +260,26 @@ private static void applyOptionsToContext(
         }
     }
 
+    private static void displayPuzzle(Puzzle puzzle, PuzzleContext ctx, PuzzleOptions options) throws IOException {
+        emitPuzzle(puzzle, ctx, options);
+
+        if (options.solutionHtml() != null) {
+            // We need a new context and a new instance of the puzzle class, so that we don't get
+            // leftover state from the first pass
+            var solveCtx = ctx.cleanCopy();
+            var puzzleForSolution = Puzzle.findByID(puzzle.id());
+            boolean solutionOutput = true;
+            applyOptionsToContext(options, solveCtx, puzzleForSolution, solutionOutput);
```

Comment:
Yeah, Java really needs parameter labels already!

_Most_ Java IDEs these days show labels for _most_ args in _most_ situations where it's helpful…but it's inconsistent and often harms readability more than it helps. I’ve configured IntelliJ to hide all of its “inlay hints” at first (that's what it calls them) but toggle them with a key shortcut when I wonder what a param is for.

Long story short, I see the original code as `solutionOutput: false` in IntelliJ, but I like your solution better: it works for everyone all the time.